v0/destinations/clickhouse/transform.js

```diff
@@ -1,12 +1,58 @@
 const { processWarehouseMessage } = require("../../../warehouse");
+const { validTimestamp } = require("../../../warehouse/util");
 
 const clickhouse = "clickhouse";
 
 function processSingleMessage(message, options) {
   return processWarehouseMessage(message, options);
 }
 
-function getDataTypeOverride(val, options) {}
+function getDataType(val) {
+  if (typeof val === "number") {
+    return Number.isInteger(val) ? "int" : "float";
+  }
+  if (typeof val === "boolean") {
+    return "boolean";
+  }
+  if (validTimestamp(val)) {
+    return "datetime";
+  }
+  return "string";
+}
+function getDataTypeOverride(val, options) {
+  // TODO: make sure all values are of same type
+  // check for [] empty arrays
+  if (Array.isArray(val)) {
+    // for now returning it as string. confirm this case
+    if (val.length === 0) {
+      return "string";
+    }
+    // check for different data types in the array. if there are different then return array(string)
+    const firstValueDataType = getDataType(val[0]);
+    let finalDataType = firstValueDataType;
+    for (let i = 1; i < val.length; i += 1) {
+      const dataType = getDataType(val[i]);
+      if (finalDataType !== dataType) {
+        if (finalDataType === "string") {
+          break;
```

Comment:
don't we need to do .toString for other types. Will clickhouse handle that

Reply:
yes, clickhouse.go is handling this